include/envoy/request_info/request_info.h

```diff
@@ -71,63 +71,123 @@ class RequestInfo {
   /**
    * @return the time that the first byte of the request was received.
    */
-  virtual SystemTime startTime() const PURE;
+  virtual const SystemTime& startTime() const PURE;
 
   /**
-   * @return duration from request start to when the entire request was received from the
-   * downstream client in microseconds.
+   * @return the monotonic time that the first byte of the request was received. Duration
+  calculations should be made relative to this value.
+  */
+  virtual const MonotonicTime& startTimeMonotonic() const PURE;
```

Comment:
Wanted to allow the caller to choose the desired duration units when performing the cast. But the same could be achieved by returning nanoseconds and allowing a caller to cast to a lower resolution unit. It would simplify things (and not require the helper utility).

Reply:
Yeah, I think if you're up for doing that it would address I wonder if the switch from SystemTime to MonotonicType addresses @mattklein123 's TODO for cleaning up time units.

Reply:
Yeah, Matt and I discussed this approach initially, but after implementing it felt really awkward to use (having to calculate the duration yourself). I'm cool with internalizing it into RequestInfo.